jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdCredentials.java

```diff
@@ -153,14 +156,27 @@ else if (!isValidType)
     {
         Fields fields = new Fields();
         fields.add("code", authCode);
-        fields.add("client_id", configuration.getClientId());
-        fields.add("client_secret", configuration.getClientSecret());
         fields.add("redirect_uri", redirectUri);
         fields.add("grant_type", "authorization_code");
+
+        Request request = configuration.getHttpClient().POST(configuration.getTokenEndpoint());
+        switch (configuration.getAuthMethod())
+        {
+            case "client_secret_basic":
+                URI uri = URI.create(configuration.getTokenEndpoint());
+                Authentication.Result authentication = new BasicAuthentication.BasicResult(uri, configuration.getClientId(), configuration.getClientSecret());
```

Comment:
Is there a test for this path?

Reply:
There is no test for this but I have tested with the google OpenID Connect services.

To write a test for this locally I have to write the OpenID provider side of it to test it against.